controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -335,9 +336,16 @@ func recordActionFailure(info *reconcileInfo, errorType metal3v1alpha1.ErrorType
 	return actionFailed{dirty: true, ErrorType: errorType, errorCount: info.host.Status.ErrorCount}
 }
 
-func recordActionDelayed(info *reconcileInfo) actionResult {
+func recordActionDelayed(info *reconcileInfo, nextState metal3v1alpha1.ProvisioningState) actionResult {
+	var counter prometheus.Counter
+
+	switch nextState {
+	case metal3v1alpha1.StateDeprovisioning, metal3v1alpha1.StateDeleting:
+		counter = delayedDeprovisioningHostCounters.With(hostMetricLabels(info.request))
+	default:
```

Comment:
I think I'd prefer to see explicitly here the cases for the provisioning counter, ie `metal3v1alpha1.StateInspecting, metal3v1alpha1.StateProvisioning`